### PR TITLE
[ShaderGraph] [Ruby] [2022.1] Adding support for triplanar in object space (or any other space..)

### PR DESCRIPTION
**Pull request by cdxntchou**
### Purpose of this PR
Fix for https://fogbugz.unity3d.com/f/cases/1346477/
Also includes an update for a failing test (inputNodes) in Linux/Vulkan platform, that is because of a trunk change to Linux/Vulkan behavior.

Documentation jira ticket: https://jira.unity3d.com/browse/GSG-685

Console PR: 🟡 -- failures match master
https://github.cds.internal.unity3d.com/unity/ScriptableRenderPipelinePrivate/pull/260

The triplanar node in normal mode has a hard-coded assumption that the input normal is in world space, and the output normal should be in tangent space.

This PR lets the user control the input and output normal spaces, and uses the standard space transform to convert the output from the input to the output space.

The default (and any existing triplanar nodes) is set to:  AbsoluteWorld =>Tangent for backwards compatibility and to match current behavior.  But the user can switch them to use Object space inputs, or World space normal outputs, as might be more appropriate depending on their use case.

Object-space triplanar (left)  vs. world-space triplanar (Right), with non-standard rotations and scales applied:
image

Triplanar World->Tangent, Triplanar Object->Tangent, Triplanar World->World, Triplanar Object->World:
image

New node controls show up in the node inspector:
Default type shows input space:
image

Normal type shows input space and normal output space:
image

---
### Testing status
- [x] Tested all the input spaces, to see that the operate as expected.
- [x] Tested Tangent, World output spaces in normal mode, generates exactly the same results when passing to the matching Normal block in the master stack.
- [x] Tested that normal results do not degrade when scaling (uniform or non) the object.

Added new graphics test to verify the new modes.

Yamato:

Nightly ShaderGraph: 🟢
https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/sg%252Ffix%252Ftriplanar-object/.yamato%252Fall-shadergraph.yml%2523Nightly_ShaderGraph_trunk/9252457/job/pipeline

https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/master/.yamato%252Fshadergraph-osx-metal.yml%2523ShaderGraph_OSX_Metal_playmode_mono_Linear_trunk/9226817/job

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Review by bencloward — APPROVED** [review]
This is going to be a great addition.

File: com.unity.shadergraph/Editor/Data/Nodes/UV/TriplanarNode.cs [hunk 1]
```diff
@@ -122,13 +168,15 @@ public virtual void GenerateNodeCode(ShaderStringBuilder sb, GenerationMode gene
...
-                        , GetVariableNameForSlot(OutputSlotId)
+                    var outputVariable = GetVariableNameForSlot(OutputSlotId);
+                    sb.AppendLine("$precision4 {0} = $precision4({1}_X.zyx * {1}_Blend.x + {1}_Y.xzy * {1}_Blend.y + {1}_Z.xyz * {1}_Blend.z, 1);"
+                        , outputVariable
                         , GetVariableNameForNode());
+
+                    // transform the normal from input to output space, and normalize
+
+                    outputVariable = $"{outputVariable}.rgb";
+                    SpaceTransformUtil.GenerateTransformCodeStatement(normalTransform, outputVariable, outputVariable, sb);
```
Comment (joshua-davis) [inline com.unity.shadergraph/Editor/Data/Nodes/UV/TriplanarNode.cs:179]: Should this have `outputVariable` for both the input and output?

Comment (cdxntchou) [reply com.unity.shadergraph/Editor/Data/Nodes/UV/TriplanarNode.cs:179]: or:  outputVariable.rgb = transform(outputVariable.rgb);   in this case

**Review by joshua-davis — APPROVED** [review]

**Comment by joshua-davis** [comment]
All tests were green before merging master for just the changelog. Cancelled re-running jobs to avoid taking up resources.

**Pull request merged**